ouroboros-network-protocols/src/Ouroboros/Network/Protocol/LocalStateQuery/Codec.hs

```diff
@@ -35,14 +35,16 @@ codecLocalStateQuery
      ( MonadST m
      , ShowQuery query
      )
-  => (point -> CBOR.Encoding)
+  => Bool   -- ^ allow 'ImmutableTip' in @'MsgAcquire'
```

Comment:
Is it possible to just pass the version instead? I see it's called in consensus and has access to the `NodeToClientVersion` or the `NodeToNodeVersion`.

Reply:
I suppose so. I primarily chose the `Bool`-approach because that's what was removed in PR 3699--just trying to match existing (albeit removed) patterns, since I'm not a codeowner here.

Reply:
I'd actually prefer to pass the version too.

Until now, when we modified a protocol we provided a new codec function or duplicated the protocol itself in some way.  Here we could generalise it and parametrise `LocalStateQuery` over the `Maybe` / `Target` type. Although this can be packaged so that it's type safe for the users, I am not sure if it's worth it.

Reply:
Done! Thanks @Jimbo4350 and @coot .

I doubt parameterizing over `Maybe` versus `Target` is worth the noise, so I've skipped that. Moreover, as Jordan commented earlier: `Nothing` and `Just` were not great names :)